frontend/src/components/NewGroup.tsx

```diff
@@ -81,38 +80,39 @@ function NewGroup(props: NewGroupProps) {
   const [selectedTab, setSelectedTab] = React.useState<number>(0); // new state
 
   let pricePerMember = 0;
-  if (splitMode === "equally" && selectedSubscription && _friends.length > 0) {
-    pricePerMember = selectedSubscription.cost / _friends.length;
-  }
-  const [customAmounts, setCustomAmounts] = useState<Record<string, number>>(
-    {}
-  );
 
-  const toast = useToast();
+  if (splitMode === "equally" && selectedSubscription && friends.length > 0) {
+    pricePerMember = parseFloat(
+      (selectedSubscription.cost / friends.length).toFixed(2)
+    );
 
-  // calculates the balance of the new subscription group
-  let subscriptionBalance: number = 0;
-  if (selectedSubscription && splitMode === "byAmount") {
-    subscriptionBalance =
-      selectedSubscription.cost -
-      Object.values(customAmounts).reduce(
-        (acc, amount) => acc + (amount || 0),
-        0
-      );
+    const updatedFriends = friends.map((friend) => ({
+      ...friend,
+      subscription_cost: pricePerMember,
+    }));
+
+    if (
+      !friends.every((friend) => friend.subscription_cost === pricePerMember)
+    ) {
+      setFriends(updatedFriends);
```

Comment:
anything with setting state shouldn't be in the component render. Either inside a useEffect, or a callback